modules/server/src/main/scala/scaladex/server/route/ProjectPages.scala

```diff
@@ -111,6 +111,37 @@ class ProjectPages(env: Env, database: WebDatabase, searchEngine: SearchEngine)(
           }
         }
       },
+
+      get {
+        path(projectM / "artifacts" ) { ref => 
+          getProjectOrRedirect(ref,user) { project => 
+            val headerF = service.getProjectHeader(project)
+            for {
+              header <- headerF
+            } yield {
+              val allArtifacts = header.toSeq.flatMap(_.artifacts)
+
+              // Group by artifact name
+              val artifactsByName = allArtifacts.groupBy(_.artifactName)
+
+              val groupedArtifacts = artifactsByName
+                .groupBy { case (artifactName, artifacts) =>  
+                  artifacts.maxBy(_.version).version 
+                }
+                .map { case (latestVersion, artifactsByName) =>
+                  latestVersion -> artifactsByName.map { case (name, artifacts) => name -> artifacts.filter(_.version == latestVersion)
+                }
+              }
+
+              val page = html.artifacts(env, user, project, header, groupedArtifacts)
```

Comment:
You will also need to pass the `params` back to the page to render the selected `binaryVersions` in the `<select>` node and to create the labels.

![image](https://github.com/user-attachments/assets/a3480c86-45eb-4fb9-9c7f-5e1ab07ea90e)